server/src/main/java/org/opensearch/gateway/TransportNodesBulkListGatewayStartedShards.java

```diff
@@ -0,0 +1,429 @@
+                storeException = null;
+            }
+            if (in.getVersion().onOrAfter(Version.V_2_3_0) && in.readBoolean()) {
+                replicationCheckpoint = new ReplicationCheckpoint(in);
+            } else {
+                replicationCheckpoint = null;
+            }
+        }
+
+        public NodeGatewayStartedShards(String allocationId, boolean primary, ReplicationCheckpoint replicationCheckpoint) {
+            this(allocationId, primary, replicationCheckpoint, null);
+        }
+
+        public NodeGatewayStartedShards(
+            String allocationId,
+            boolean primary,
+            ReplicationCheckpoint replicationCheckpoint,
+            Exception storeException
+        ) {
+            this.allocationId = allocationId;
+            this.primary = primary;
+            this.replicationCheckpoint = replicationCheckpoint;
+            this.storeException = storeException;
+        }
+
+        public String allocationId() {
+            return this.allocationId;
+        }
+
+        public boolean primary() {
+            return this.primary;
+        }
+
+        public ReplicationCheckpoint replicationCheckpoint() {
+            return this.replicationCheckpoint;
+        }
+
+        public Exception storeException() {
+            return this.storeException;
+        }
+
+        public void writeTo(StreamOutput out) throws IOException {
+            TransportNodesGatewayStartedShardHelper.NodeGatewayStartedShardsWriteTo(
+                out,
+                allocationId,
+                primary,
+                storeException,
+                replicationCheckpoint
+            );
+        }
+
+        @Override
+        public boolean equals(Object o) {
+            if (this == o) {
+                return true;
+            }
+            if (o == null || getClass() != o.getClass()) {
+                return false;
+            }
+
+            NodeGatewayStartedShards that = (NodeGatewayStartedShards) o;
+
+            return primary == that.primary
+                && Objects.equals(allocationId, that.allocationId)
+                && Objects.equals(storeException, that.storeException)
+                && Objects.equals(replicationCheckpoint, that.replicationCheckpoint);
+        }
+
+        @Override
+        public int hashCode() {
+            return TransportNodesGatewayStartedShardHelper.NodeGatewayStartedShardsHashCode(
+                allocationId,
+                primary,
+                storeException,
+                replicationCheckpoint
+            );
+        }
+
+        @Override
+        public String toString() {
+            return TransportNodesGatewayStartedShardHelper.NodeGatewayStartedShardsToString(
+                allocationId,
+                primary,
+                storeException,
+                replicationCheckpoint
+            );
+        }
+    }
+
+    public static class BulkOfNodeGatewayStartedShards extends BaseNodeResponse {
```

Comment:
Better name ? 

`NodeGatewayStartedShardCollection` ?
or 

`NodeGatewayStartedShardsBatch`

Reply:
Renamed to NodeGatewayStartedShardsBatch